src/main/java/de/craften/ui/swingmaterial/MaterialPanel.java

```diff
@@ -5,36 +5,80 @@
 import java.awt.*;
 
 /**
- * A material panel with elevation.
+ * A JPanel customized for Material components. What makes these panels special
+ * is the possibility of assigning them an elevation level. Elevation can help
+ * distinguishing elements inside a Material-based GUI, and any changes done to
+ * them result in nicely animated transitions, helping to achieve that Material
+ * flow.
+ * <p/>
+ * However, there is a catch: shadows are kinda expensive to compute, as
+ * Java2D relies on the CPU for anything other than images, so having a lot
+ * of elements with a given elevation (and thus, a shadow) can reduce
+ * performance when these elevations change because of the triggered animations.
+ * <p/>
+ * Letting the components suggest a prefered size based on their contents is
+ * still under development, so it is not advised to use your favorite
+ * {@link LayoutManager} inside a {@code MaterialPanel} unless you set the
+ * prefered, minimum and maximum size of each component by your self. Currently,
+ * the prefereable approach to follow is overriding {@link #doLayout()} and
+ * taking care of any arrangements by yourself.
  */
 public class MaterialPanel extends JPanel {
     private final ElevationEffect elevation;
 
     /**
-     * Creates a new panel.
+     * Creates a new {@code MaterialPanel}. For the moment, this is the only
+     * supported constructor from JPanel.
      */
     public MaterialPanel() {
+        super();
         elevation = ElevationEffect.applyTo(this, 1);
         setBorder(new EmptyBorder(MaterialShadow.OFFSET_TOP, MaterialShadow.OFFSET_LEFT, MaterialShadow.OFFSET_BOTTOM, MaterialShadow.OFFSET_RIGHT));
     }
 
     /**
-     * Gets the elevation level.
+     * Gets the elevation level of this panel. Changes in elevation trigger an
+     * animated transition if the component is currently visible, so it is
+     * incorrect to assume the returned value will reflect how the resulting
+     * shadow looks right now.
      *
      * @return elevation level (0..5)
+     * @see ElevationEffect
      */
     public int getElevation() {
         return elevation.getLevel();
     }
 
     /**
-     * Sets the elevation level.
+     * Sets the elevation level of this panel. Changes in elevation trigger an
+     * animated transition if the component is currently visible, so it will
+     * take a little while for the resulting shadow to reflect the level once
+     * it is set.
      *
      * @param elevation elevation level (1..5)
```

Comment:
Whoops, I documented the param wrong here, the elevation level is of course 0..5, as above. Would you like to fix that, @DragShot?